scripts/cot/botania/flowers/genereting/nuclianthus.zs

```diff
@@ -0,0 +1,233 @@
+/*
+Nuclianthus (Helianthus + nuclear) - botanical reactor
+Generates power using nuclear fuel (produces nuclear waste)
+Catch: if buffer is overflowing (for some mana amount wasted - but heals overtime) it explodes KABOOM!!!
+
+Stats:
+Mana per 1 sec: H/t*eff
+Time working: as on fuel
+*/
+#loader contenttweaker
+
+import crafttweaker.data.IData;
+import crafttweaker.entity.IEntityItem;
+import crafttweaker.item.IItemStack;
+import crafttweaker.oredict.IOreDict;
+import crafttweaker.util.Math;
+import crafttweaker.world.IBlockPos;
+import crafttweaker.world.IWorld;
+import mods.contenttweaker.Color;
+import mods.contenttweaker.VanillaFactory;
+import mods.randomtweaker.cote.BlockAdded;
+import mods.randomtweaker.cote.ISubTileEntity;
+import mods.randomtweaker.cote.ISubTileEntityFunctional;
+import mods.randomtweaker.cote.ISubTileEntityGenerating;
+import mods.randomtweaker.cote.SubTileEntityInGame;
+import mods.zenutils.DataUpdateOperation.OVERWRITE;
+
+static fuelsList as int[][string] = {
+/*
+Fromat:
+fuel:name [duration, heat/t, effciency]
+*/
+
+// Thorium fuels
+'item.nuclearcraft.fuel_thorium.tbu_ox': [14400, 40, 125], // TBU-OX
+'item.nuclearcraft.fuel_thorium.tbu_ni': [18000, 32, 125], // TBU-NI
+'item.nuclearcraft.fuel_thorium.tbu_za': [11520, 50, 125],  // TBU-ZA
```

Comment:
Having this type of pedefined values will make a "correcting bomb", when I will need to remember every place in code to correct values if NC configs is changed.
It will be even worst since NC is going to be updated soon, so configuration and default values may change.
Please, check out if NC allowing to have native getters. Maybe you could just call a function like `native.nuclearcraft.fuels.getEfficiency()` or something like that.

Reply:
I'll try to check it

Reply:
I've managed to get to the class of fuel with native methods and get class of ItemFissionFuel
https://github.com/tomdodd4598/NuclearCraft/blob/1.12.2o/src/main/java/nc/item/ItemFissionFuel.java

however to get stats of fuel i need to get IFissionFuelEnum. I can't access it, i get instance of java.lang.Class with is sensitive in zs :( Looks like I'm stuck here